mkosi/__init__.py

```diff
@@ -625,6 +625,19 @@ def btrfs_subvol_make_ro(path: str, b: bool = True) -> None:
     run(["btrfs", "property", "set", path, "ro", "true" if b else "false"])
 
 
+@contextlib.contextmanager
+def btrfs_forget_stale_devices(args: CommandLineArguments) -> Generator[None, None, None]:
+    # When using cached images (-i), mounting btrfs images would sometimes fail
+    # with EEXIST. This is likely because a stale device is leftover somewhere
+    # from the previous run. To fix this, we make sure to always clean up stale
+    # btrfs devices after unmounting the image.
+    try:
+        yield
+    finally:
+        if shutil.which("btrfs"):
```

Comment:
If we don't ask for `args.output_format == OutputFormat.gpt_btrfs` here as well, we can omit the argument in the signature, no point in passing it, if it's not used.

Reply:
Nice catch, forgot to remove that.